src/lib/operators/insert.cpp

```diff
@@ -251,8 +251,6 @@ void Insert::_on_rollback_records() {
      * foreign tid - which is what a visible row that is being deleted by a different transaction looks like. Thus,
      * the other transaction would consider the row (that is in the process of being rolled back and should have never
      * been visible) as visible.
-     *
-     * We need to set `begin_cid = 0` so that the ChunkCompressionTask can identify "completed" Chunks.
      */
 
     for (auto chunk_offset = target_chunk_range.begin_chunk_offset; chunk_offset < target_chunk_range.end_chunk_offset;
```

Comment:
We should be able to merge the following two for-loops into one.

Reply:
I lack the understanding to judge on that. I guess we have tests in place that would recognize if we cannot merge the loops?

Reply:
Hmm, not sure about these fences though. :-/

Reply:
After removing `begin_cid = 0` and merging both for-loops, the first fence should become irrelevant because `begin_cid = MAX_COMMIT_ID` is always true.

From my understanding, this leaves the possibility that other threads can see rows as follows:

* `end_cid = 0` & `tid = 0` (desired result after rollback)
* `end_cid = MAX_COMMIT_ID` & `tid = 0`
* `end_cid = 0` & `tid = CurrentTransactionID`
* `end_cid = MAX_COMMIT_ID` & `tid = CurrentTransactionID`


These options do not look critical to me since other transactions should not see these rows anyway (thanks to `begin_cid = MAX_COMMIT_ID`).

I'm not sure if we need the last memory fence though.

Reply:
CI passes. Do you see any issues @mrks?

Reply:
From the first look, this should be fine. I'll have to find more than just a free minute to think about it though.